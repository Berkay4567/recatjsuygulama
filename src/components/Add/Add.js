import React from 'react'
import styles from "./Add.module.scss";
import { Link } from "react-router-dom";
const Add = () =>{
    return(
        <Link  to={"/addnew"}>
        <div>
            <button className={styles.button}> + </button>
        </div>
        </Link>
    )
}

export default Add