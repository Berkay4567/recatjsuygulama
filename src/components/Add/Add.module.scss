.button{
    display: inline-block;
    padding: 0px 30px;
    font-size: 70px;
    cursor: pointer;
    text-align: center;
    text-decoration: none;
    outline: none;
    color: #fff;
    background-color: #3178d6;
    border: none;
    border-radius: 50%;
    box-shadow: 0 9px #999;
}
.button2{
    display: inline-block;
    padding: 0px 30px;
    font-size: 70px;
    cursor: pointer;
    text-align: center;
    text-decoration: none;
    outline: none;
    color: #fff;
    background-color: #1fc91f;
    border: none;
    border-radius: 50%;
    box-shadow: 0 9px #999;
}