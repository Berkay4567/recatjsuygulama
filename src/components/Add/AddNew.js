import React,{ useState} from 'react';
import { Link } from "react-router-dom";
const AddNew=()=>{
	let url = `https://5fc9346b2af77700165ae514.mockapi.io/simpsons`;
	
	//---Tanımlar----
	const[name,setName]=useState("");
	const[job,setJob]=useState("");
	const[about,setAbout]=useState("");
	const[avatar,setAvatar]=useState("");
	//---------------

    return(
		<div className="container">
			<div>
				<h1 className="text-center mt-5">Add a new Simpsons</h1>
				<form>
					<div className="form-group">
						<label>Name Surname:</label>
						<input
							type="name"
							className="form-control"
							placeholder="Full Name"
							onChange={e => setName(e.target.value)}
						/>
					</div>
					<div className="form-group">
						<label>Job Title:</label>
						<input
							type="job"
							className="form-control"
							placeholder="Enter job title"
							onChange={e => setJob(e.target.value)}
						/>
					</div>
					<div className="form-group">
						<label>About Him/Her:</label>
						<input
							type="about"
							className="form-control"
							placeholder="Enter about him/her"
							onChange={e => setAbout(e.target.value)}
						/>
					</div>
					<div className="form-group">
						<label>Image Link</label>
						<input
							type="img"
							className="form-control"
							placeholder="Enter image link"
							onChange={e => setAvatar(e.target.value)}
						/>
					</div>
                    <br/>
					<Link to={"/"}>
						<button
							type="button"
							className="btn btn-primary form-control"
							onClick={()=>ekle(name, job, about, avatar)}
						>
							Add Character
						</button>
					</Link>
					<Link className="mt-3 w-100 text-center" to="/" >
						or get back to contacts
					</Link>
				</form>
			</div>
		</div>

    )
	//------------------EKLEME KOMUTU-----------------------
	function ekle(name, job, about, avatar){
		try{
			return fetch(url,{
				method:"post",
				headers:{"Content-type": "application/json"},
				body: JSON.stringify({
					name:name,
					job:job,
					about:about,
					avatar:avatar
				})
			})
			.then(response => response.json())
			.then(data => {
				console.log("data",data)
	
			})
			
	
		}
		catch(err){
			console.log(err);
		}
	}
	//------------------EKLEME KOMUTU BİTİMİ------------------
}

export default AddNew