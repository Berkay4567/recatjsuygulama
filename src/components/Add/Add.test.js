import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router} from 'react-router-dom';
import Add from './Add';

it("renders when there are no items", ()=>{
    const tree=renderer.create(<Router>
        <Add />
      </Router>
  ).toJSON();
    expect(tree).toMatchSnapshot();
})