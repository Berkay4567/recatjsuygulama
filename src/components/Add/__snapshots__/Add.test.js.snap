// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are no irems 1`] = `
<a
  href="/addnew"
  onClick={[Function]}
>
  <div>
    <button
      className="button"
    >
       + 
    </button>
  </div>
</a>
`;

exports[`renders when there are no items 1`] = `
<a
  href="/addnew"
  onClick={[Function]}
>
  <div>
    <button
      className="button"
    >
       + 
    </button>
  </div>
</a>
`;
